src/test/java/twitterapp/src/ResourceTest.java

```diff
@@ -36,57 +43,79 @@ public void setUp() {
-        assertEquals("Tweet("+shortTweet+") has been posted",r.getEntity().toString());
+        assertEquals(Response.Status.OK, Response.Status.fromStatusCode(r.getStatus()));
+        assertEquals("Tweet("+shortTweet+") has been posted.",r.getEntity().toString());
     }
 
+    @Test
+    public void testFailTweet(){
+        try {
+            String tweet = "test tweet";
+            when(mockTwitter.updateStatus(tweet)).thenThrow(new TwitterException("There was a problem on the server side, please try again later."));
+            r = resource.postTweet(tweet);
+            assertEquals(Response.Status.INTERNAL_SERVER_ERROR, Response.Status.fromStatusCode(r.getStatus()));
+        }
+        catch(Exception e){
+            fail();
+        }
+    }
 
     @Test
     public void testLongTweet(){
         String longTweet = StringUtils.repeat(".",MAX_LENGTH + 1);
         r = resource.postTweet(longTweet);
-        assertEquals(500, r.getStatus());
-        assertEquals("Tweet is too long, keep it with in 280 characters",r.getEntity().toString());
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR, Response.Status.fromStatusCode(r.getStatus()));
+        assertEquals("Tweet is too long, keep it within 280 characters",r.getEntity().toString());
     }
 
     @Test
     public void testEmptyTweet(){
         String emptyTweet = "";
         r = resource.postTweet(emptyTweet);
-        assertEquals(500, r.getStatus());
-        assertEquals("No tweet entered.",r.getEntity());
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR, Response.Status.fromStatusCode(r.getStatus()));
+        assertEquals("No tweet entered",r.getEntity());
     }
 
     @Test
     public void testEmptyTimeline(){
         r = resource.getTimeline();
         when(r.getEntity()).thenReturn(null);
-        assertEquals(500, r.getStatus());
-        assertEquals(MediaType.APPLICATION_JSON_TYPE, r.getMediaType());
+        assertEquals(Response.Status.INTERNAL_SERVER_ERROR, Response.Status.fromStatusCode(r.getStatus()));
+        assertEquals("There was a problem on the server side, please try again later.",r.getEntity().toString());
     }
 
     @Test
-    public void testTimelineReturnJSON(){
+    public void testTimelineReturnJSON() {
         ResponseList<Status> responseList = new TwitterResponseList<Status>();
-    ;
         Status mockStatus = mock(Status.class);
-         try {
-             when(mockStatus.getText()).thenReturn("mockStatus");
-             responseList.add(mockStatus);
-             when(mockTwitter.getHomeTimeline()).thenReturn(responseList);
-             r = resource.getTimeline();
-             ResponseList<Status> newResponseList = (ResponseList<Status>) r.getEntity();
-             assertEquals(1,newResponseList.size());
-             assertEquals("mockStatus", responseList.get(0).getText());
+        try {
+            when(mockStatus.getText()).thenReturn("mockStatus");
+            responseList.add(mockStatus);
```

Comment:
have multiple items in the list with different messages and verify they match up, this allows you to also test the ordering of items returned matches as well